program-tests/nft-create-v2.test.ts

```diff
@@ -0,0 +1,182 @@
+import * as anchor from "@project-serum/anchor";
+import { AnchorProvider, Program } from "@project-serum/anchor";
+import { Keypair, SystemProgram } from "@solana/web3.js";
+import { createCollection } from "./utils/create-collection";
+import { createNftV2 } from "./utils/create-nft";
+import { KeypairWallet } from "./utils/KeypairWallet";
+import {
+  airdropSol,
+  DEFAULT_KEYPAIR,
+  NftokenIdl,
+  NULL_PUBKEY_STRING,
+  program,
+  PROGRAM_ID,
+} from "./utils/test-utils";
+
+describe("ix_nft_create_v2", () => {
+  const provider = anchor.AnchorProvider.env();
+  anchor.setProvider(provider);
+
+  test("mints an NFT", async () => {
+    const authority = DEFAULT_KEYPAIR.publicKey;
+    const { nft_pubkey } = await createNftV2({});
+    const nft = await program.account.nftAccount.fetch(nft_pubkey);
+
+    expect(nft.authority.toBase58()).toEqual(authority.toBase58());
+    expect(nft.holder.toBase58()).toEqual(authority.toBase58());
+    expect(nft.collection.toBase58()).toEqual(NULL_PUBKEY_STRING);
+    expect(nft.version).toEqual(1);
+  });
+
+  test("mints an NFT to a different holder", async () => {
+    const authority = DEFAULT_KEYPAIR.publicKey;
+    const holder = Keypair.generate().publicKey;
+    const { nft_pubkey } = await createNftV2({ holder });
+    const nft = await program.account.nftAccount.fetch(nft_pubkey);
+
+    expect(nft.authority.toBase58()).toEqual(authority.toBase58());
+    expect(nft.holder.toBase58()).toEqual(holder.toBase58());
+    expect(nft.collection.toBase58()).toEqual(NULL_PUBKEY_STRING);
+    expect(nft.version).toEqual(1);
+  });
+
+  test("mints an NFT with a different payer", async () => {
```

Comment:
This test case is the de facto difference between `nft-create-v1.test.ts` and `nft-create-v2.test.ts`.

Reply:
Thanks! Looks good 👍